sqlglot/expressions.py

```diff
@@ -36,6 +35,7 @@
 from sqlglot.tokens import Token, TokenError
 
 if t.TYPE_CHECKING:
+    from typing_extensions import Self
```

Comment:
By quoting + putting this under a typing-extensions block, we can still utilize backported typing features while keeping compatibility for those who don't having typing-extensions installed.

Reply:
Should we do this for Query methods too and get rid of the `Q` type variable defined below?

Reply:
That would probably be good, would you prefer I do this in this PR, or happy to follow-up with this?

Reply:
I'm fine with both, your call 👍

Reply:
I'll follow-up in that case, just to keep the focus of this PR.